src/reducers/usersReducer.js

```diff
@@ -0,0 +1,38 @@
+import initialState from '../store/initialState';
+import { REGISTER_REQUEST, REGISTER_FAILURE, REGISTER_SUCCESS } from '../actions/actionTypes';
+
+const { user } = initialState;
+const userReducer = (state = user, action) => {
+  const { type } = action;
+  switch (type) {
+    case REGISTER_REQUEST: {
+      return {
+        ...state,
+        isLoading: true,
+        fullNameError: [],
+        emailError: []
+      };
+    }
+    case REGISTER_FAILURE: {
+      return {
+        ...state,
+        fullNameError: action.error.fullName,
+        emailError: action.error.email,
+        isLoading: false
+      };
+    }
+    case REGISTER_SUCCESS: {
+      return {
+        ...state,
+        isLoading: false,
+        fullNameError: [],
+        emailError: [],
+        isSuccess: true
+      };
+    }
+    default:
+      return state;
```

Comment:
When this returns, wouldn't the new state only contain the user object?

Reply:
Yes, it should contain only the user object because the userReducer only make reference to the user object.